src/audio/base_fw_intel.c

```diff
@@ -224,8 +224,8 @@ static uint32_t basefw_get_ext_system_time(uint32_t *data_offset, char *data)
 	ext_system_time_data.rtc_l = (uint32_t)rtc;
 	ext_system_time_data.rtc_u = (uint32_t)(rtc >> 32);
 
-	memcpy_s(ext_system_time, sizeof(ext_system_time), &ext_system_time_data,
-		 sizeof(ext_system_time));
+	memcpy_s(ext_system_time, sizeof(struct ipc4_ext_system_time), &ext_system_time_data,
+		 sizeof(struct ipc4_ext_system_time));
```

Comment:
I guess "sizeof(ext_system_time_data)" was the original intent ("sizeof(*ext_system_time)" would have done as well). A good catch nevertheless. This is not used by the Linux host so was not caught with Linux driver tests.

Reply:
yes, a very good catch, but indeed a
```
memcpy_s(ext_system_time, sizeof(*ext_system_time), &ext_system_time_data,
		 sizeof(ext_system_time_data));
```
would be even better IMHO.

Reply:
@lyakh Good point, I changed this line. Thank you